### PR TITLE
feat(pillar): some config values can be treated as lists

### LINKED ISSUE #14: [FEATURE] Treat other_hostnames and local_interfaces as list

### PR DESCRIPTION
**Pull request by n-rodriguez**
Fix https://github.com/saltstack-formulas/exim-formula/issues/14

### CODE REVIEW

File: test/salt/pillar/exim.sls [hunk 1]
```diff
@@ -3,8 +3,13 @@
 ---
 exim:
   config:
-    configtype: 'satellite'
+    configtype: satellite
     hide_mailname: 'false'
```
Comment (vutny) [inline test/salt/pillar/exim.sls:7]: You probably want to make `hide_mailname` value as boolean  too.

**Review by vutny — APPROVED** [review]
Awesome, @n-rodriguez . I love the `join_if_list` macro!
There is a small suggestion below, but you got my approval.

File: test/salt/pillar/exim.sls (hunk 1 above)
Comment (n-rodriguez) [reply test/salt/pillar/exim.sls:7]: > You probably want to make `hide_mailname` value as boolean too.

I want to test that string values and boolean values will be casted in the same way.

Comment (vutny) [reply test/salt/pillar/exim.sls:7]: Sounds reasonable!

**Comment by n-rodriguez** [comment]
> Awesome, @n-rodriguez . I love the `join_if_list` macro!

Actually it's not from me but from https://github.com/jerrykan/exim-formula

https://github.com/jerrykan/exim-formula/blob/master/exim/files/update-exim4.conf.conf.jinja

**Comment by aboe76** [comment]
maybe @jerrykan would like to help too?

**Review by aboe76 — APPROVED** [review]

**Pull request merged**